app/src/main/java/space/taran/arknavigator/ui/fragments/GalleryFragment.kt

```diff
@@ -143,99 +153,111 @@ class GalleryFragment(
+                notifyUser("Tag \"$tag\" removed")
+                presenter.onTagRemove(tag)
+                true
+            }
 
-        if (resources.isEmpty()) {
-            presenter.quit()
+            binding.tagsCg.addView(chip)
         }
+
+        binding.tagsCg.addView(createEditChip())
     }
 
-    override fun showEditTagsDialog(position: Int) {
-        val resource = resources[position]
+    override fun showEditTagsDialog(
+        resource: ResourceId
+    ) {
         Log.d(GALLERY_SCREEN, "showing [edit-tags] dialog for resource $resource")
-        val dialog = EditTagsDialogFragment(resource.id, storage, index, ::onTagsChanged)
-        dialog.show(childFragmentManager, dialog.TAG)
+        val dialog = EditTagsDialogFragment.newInstance(
+            requireArguments()[ROOT_AND_FAV_KEY] as RootAndFav,
+            resource
+        )
+        dialog.show(childFragmentManager, EditTagsDialogFragment.TAG)
+        dialog.onTagsChangedListener = presenter::onTagsChanged
     }
 
-    private fun setupOpenEditFABs() {
-        val position = binding.viewPager.currentItem
+    override fun backClicked(): Boolean {
+        Log.d(GALLERY_SCREEN, "[back] clicked in GalleryFragment")
+        return presenter.onBackClick()
+    }
 
-        binding.apply {
-            openResourceFab.setOnClickListener(null)
-            editResourceFab.setOnClickListener(null)
+    private fun setTitle(title: String) {
+        (requireActivity() as MainActivity).setTitle(title)
+    }
 
-            when (resources[position].kind) {
+    private fun setupOpenEditFABs(kind: ResourceKind?) {
+        binding.apply {
+            when (kind) {
                 ResourceKind.VIDEO -> {
                     // "open" capabilities only
                     editResourceFab.makeGone()
-                    openResourceFab.makeVisibleAndSetOnClickListener {
-                        viewInExternalApp(position)
-                    }
+                    openResourceFab.makeVisible()
                 }
                 ResourceKind.DOCUMENT -> {
                     // both "open" and "edit" capabilities
-                    editResourceFab.makeVisibleAndSetOnClickListener {
-                        editResource(position)
-                    }
-
-                    openResourceFab.makeVisibleAndSetOnClickListener {
-                        viewInExternalApp(position)
-                    }
+                    editResourceFab.makeVisible()
+                    openResourceFab.makeVisible()
                 }
                 ResourceKind.IMAGE -> {
                     // "edit" capabilities only
                     openResourceFab.makeGone()
-                    editResourceFab.makeVisibleAndSetOnClickListener {
-                        editResource(position)
-                    }
+                    editResourceFab.makeVisible()
+                }
+                null -> {
+                    //TODO
```

Comment:
?

Reply:
What should be the visibility of opening and editing fabs if the resource kind is not defined?

Reply:
Only opening would be fine, also share button.